src/grants/models/grant.js

```diff
@@ -185,20 +188,43 @@ export class Grant {
     return status;
   }
 
-  #isValidFromMatch(validFrom, currentStatus) {
-    return validFrom.some((validFromStatus) => {
-      if (validFromStatus.includes(":")) {
-        // Fully qualified status like "PRE_AWARD:REVIEW_APPLICATION:APPROVED"
-        return validFromStatus === currentStatus;
+  #normaliseValidFrom(statusDef) {
+    const statusLevelProcesses = Array.isArray(statusDef.processes)
+      ? statusDef.processes
+      : [];
+
+    return (statusDef.validFrom || []).map((v) => {
+      if (typeof v === "string") {
+        return { code: v, processes: statusLevelProcesses };
       }
-      // Simple status code - extract just the status part from currentStatus
-      const currentStatusCode = currentStatus.includes(":")
-        ? currentStatus.split(":").pop()
-        : currentStatus;
-      return validFromStatus === currentStatusCode;
+      return {
+        code: v.code,
+        processes: Array.isArray(v.processes) ? v.processes : [],
+      };
     });
   }
 
+  // eslint-disable-next-line complexity
+  #findMatchingValidFromEntry(validFromEntries, currentStatus) {
+    const currentStatusCode = currentStatus.includes(":")
```

Comment:
you don't actually need the check here - split will return an array of 1 if there's no colon
this will return STATUS from any of the following ```"PHASE:STAGE:STATUS", "::STATUS", "STATUS"```
```suggestion
    const currentStatusCode = currentStatus.split(":").pop();
```